driver-sync/src/main/com/mongodb/client/internal/MongoClientDelegate.java

```diff
@@ -183,7 +185,13 @@ private ClusterDescription getConnectedClusterDescription() {
             WriteBinding binding = getWriteBinding(readConcern, actualClientSession, session == null && actualClientSession != null);
             try {
                 return operation.execute(binding);
-            } finally {
+            } catch (MongoSocketException e) {
+                if (session != null && session.hasActiveTransaction()) {
+                    e.addLabel(MongoException.TRANSIENT_TRANSACTION_ERROR_LABEL);
```

Comment:
I also decided that it was ok to make MongoException mutable with this addLabel method.  The justification is that exception generally aren't shared so mutability isn't that bad.    There is also precedent with the java.lang.Throwable#initCause method.